packages/form-field/src/FormFieldFeedback/FormFieldFeedback.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { color } from '@leafygreen-ui/tokens';
+import { Body, Error, useUpdatedBaseFontSize } from '@leafygreen-ui/typography';
+
+import { LGIDS_FORM_FIELD } from '../constants';
+import { getFontSize } from '../FormField/FormField.styles';
+import { FormFieldState } from '../FormField/FormField.types';
+
+import {
+  containerStyles,
+  getIconWrapperHeight,
+  hideContainerStyle,
+  iconWrapperStyles,
+  spacingTop,
+} from './FormFieldFeedback.styles';
+import { FormFieldFeedbackProps } from './FormFieldFeedback.types';
+
+export const FormFieldFeedback = ({
+  disabled,
+  errorMessage,
+  hideFeedback = false,
+  id,
+  size,
+  state,
+  successMessage,
+  ...rest
+}: FormFieldFeedbackProps) => {
+  const { theme } = useDarkMode();
+  const baseFontSize = useUpdatedBaseFontSize();
+  const fontStyles = getFontSize({ baseFontSize, size });
+
+  const isErrorState = state === FormFieldState.Error;
+  const isValidState = state === FormFieldState.Valid;
+  const showFormFieldFeedback = (isErrorState || isValidState) && !disabled;
+
+  const iconProps = {
+    glyph: isErrorState ? 'Warning' : 'Checkmark',
+    fill: color[theme].icon[isErrorState ? 'error' : 'success'].default,
```

Comment:
Could we just refer directly to the `state` property? 
 `color[theme].icon[state].default`

Reply:
https://github.com/mongodb/leafygreen-ui/pull/2318/commits/3b444b2d490b74792fc8d9972616b1d6681d4db1#diff-c731bce236af5cb475ed9909a79e3033cc4fd1d9119accb38c1d426b377f70eaR41-R52
couldn't just directly key into `state` because `FormFieldState` uses `'valid'` while color variant uses `'success'`. made a style util that converts the `FormFieldState` into a color variant